### PR TITLE
update hammerdb 3.3 to 4.0 and adapt the result parsing to hammerdb4.0

### PR DESCRIPTION
**Pull request by ebattat**
### Description
Update hammerdb version from 3.3 to 4.0
### Dependency
** This PR is depend on benchmark-operator hammerdb fix:
https://github.com/cloud-bulldozer/benchmark-operator/pull/569

### Fixes
1. Dockerfile
Install hammerdb 4.0 and update new hammerdb download url

2. trigger_hammerdb.py 
Change parsing way in hammerdb 4.0 and also cast to integer that elastic search will recognize it as number
Change timestamp from string that elastic search will recognize it as timestamp

Depends-On: 569

### CODE REVIEW

**Review by mkarg75 — APPROVED** [review]
LGTM

**Review by amitsagtani97 — CHANGES_REQUESTED** [review]
hey @ebattat , please run the pre-commit to fix the code lint issues [https://github.com/cloud-bulldozer/benchmark-wrapper/blob/master/README.md#codestyling-and-linting]

**Comment by ebattat** [comment]
It failed because need to rebuild hammerdb Dockerfile and upload it to: 
quay.io/cloud-bulldozer/hammerdb:master
Who can do it, I dont have permissions for it ?

**Comment by amitsagtani97** [comment]
> It failed because need to rebuild hammerdb Dockerfile and upload it to:
> quay.io/cloud-bulldozer/hammerdb:master
> Who can do it, I dont have permissions for it ?

For the CI, we build a new image every time to test it 
`19:54:41 + podman build --no-cache --tag=quay.io/rht_perf_ci/hammerdb:snafu_ci -f snafu/hammerdb/Dockerfile .
`
It failed due to timeout issue here 
`19:58:04 ++ kubectl wait --for=condition=complete -l app=hammerdb_workload-e27def64 --namespace my-ripsaw-32d9b291-887b-4d19-a196 jobs --timeout=500s
20:06:24 error: timed out waiting for the condition on jobs/hammerdb-workload-e27def64`

Complete logs - https://ec2-52-39-240-23.us-west-2.compute.amazonaws.com:8443/job/snafu/319/console

Do you think we will need more time for the completion the test with these changes?

and regarding pre-commit its reporting error due to unused `import time` statement in the trigger_hammerdb.py

**Comment by ebattat** [comment]
Hi,

I have fixed import datetime issue.
Can you please try it again ?

Eli.

**Comment by ebattat** [comment]
Hi,

Maybe we need to enlarge the timeout to 1000s.
What do you think ?

**Comment by amitsagtani97** [comment]
> Hi,
> 
> I saw this error before the timeout issue:
> 14:24:41 Error: unable to get images to prune: error reading image "bff5dedea21509112754102d747397f986c61b85aafdf06ca1371fe3b78b360b" as image: error locating item named "manifest" for image with ID "bff5dedea21509112754102d747397f986c61b85aafdf06ca1371fe3b78b360b": file does not exist
> 
> Looks like issue with pulling the new image. can you please take a look ?
> 
> Eli.

This pr depends on this one  - https://github.com/cloud-bulldozer/benchmark-operator/pull/569 , right?

**Comment by ebattat** [comment]
Hi,

Yes, also mentioned it in PR cloud-bulldozer/benchmark-operator#569
Do you think that cause the issue ?

Eli.

**Comment by amitsagtani97** [comment]
/rerun all

> Hi,
> 
> Yes, also mentioned it in PR [cloud-bulldozer/benchmark-operator#569](https://github.com/cloud-bulldozer/benchmark-operator/pull/569)
> Do you think that cause the issue ?
> 
> Eli.

Yes, we need to add "`Depends-On: <benchmark-operator_pr_number>`" in the pr description to create image from there, as mentioned in https://github.com/cloud-bulldozer/benchmark-wrapper/blob/master/README.md#how-do-i-run-my-snafu-wrapper-in-ci 
Updated the pr description :+1: 
Giving it a retry now.

**Comment by ebattat** [comment]
Hi,

Can you share the logs ?

Eli.

**Comment by amitsagtani97** [comment]
> Hi,
> 
> Can you share the logs ?
> 
> Eli.

  - https://ec2-52-39-240-23.us-west-2.compute.amazonaws.com:8443/job/snafu/322/
  
  We can ignore the podman pruner warning for now.. something seems to be wrong with the benchmark-operator pr.. need to test things locally for more verbose logs.

**Comment by ebattat** [comment]
Hi,

I have updated hammer mssql test in benchmark operator.
Can you please try to rerun with my benchmark operator branch: https://github.com/cloud-bulldozer/benchmark-operator/pull/569

Thanks,
Eli.

File: snafu/hammerdb/trigger_hammerdb.py [hunk 1]
```diff
@@ -94,9 +94,10 @@ def _parse_stdout(self, stdout):
         data = []
         for line in stdout.splitlines():
             if "TEST RESULT" in line:
-                worker = (line.split(":"))[0]
-                tpm = (line.split(" "))[6]
-                nopm = (line.split(" "))[-2]
+                worker_name = (line.split())[1]
+                worker = int((worker_name.split(":"))[0])
+                tpm = int((line.split())[-3])
```
Comment (amitsagtani97) [inline snafu/hammerdb/trigger_hammerdb.py:99]: got this error while testing it locally

`Traceback (most recent call last):
  File "/usr/local/bin/run_snafu", line 33, in <module>
    sys.exit(load_entry_point('snafu', 'console_scripts', 'run_snafu')())
  File "/opt/snafu/snafu/run_snafu.py", line 118, in main
    for i in process_generator(index_args, parser):
  File "/opt/snafu/snafu/run_snafu.py", line 138, in process_generator
    for action, index in data_object.emit_actions():
  File "/opt/snafu/snafu/hammerdb/trigger_hammerdb.py", line 236, in emit_actions
    data = self._parse_stdout(stdout[0])
  File "/opt/snafu/snafu/hammerdb/trigger_hammerdb.py", line 99, in _parse_stdout
    tpm = int((line.split())[-3])
ValueError: invalid literal for int() with base 10: 'SQL'`

**Review by amitsagtani97 — CHANGES_REQUESTED** [review]

File: snafu/hammerdb/trigger_hammerdb.py (hunk 1 above)
Comment (amitsagtani97) [reply snafu/hammerdb/trigger_hammerdb.py:99]: (worker_name.split(":"))[0]  return 'SQL' I guess

Comment (amitsagtani97) [reply snafu/hammerdb/trigger_hammerdb.py:99]: `(line.split())[-3] ` return 'SQL'

Comment (ebattat) [reply snafu/hammerdb/trigger_hammerdb.py]: Thx Amit, Fix it in the last commit. (parse according to database type)

**Comment by jtaleric** [comment]
@amitsagtani97 you good with this change?

**Review by amitsagtani97 — APPROVED** [review]
LGTM!

**Comment by jtaleric** [comment]
Needs a rebase.

**Comment by ebattat** [comment]
Hi,

I have rebased both: benchmark wrapper
<https://github.com/cloud-bulldozer/benchmark-operator/pull/569/commits/13681862c6a8efe692dc716631f6beb450e0c76f>
and benchmark operator
<https://github.com/cloud-bulldozer/benchmark-operator/pull/569/commits/13681862c6a8efe692dc716631f6beb450e0c76f>
.
Please rerun hammerdb CI.

Best regards,
Eli Battat.

**Comment by jtaleric** [comment]
I still see 
```
This branch cannot be rebased due to conflicts
Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.
```

**Comment by ebattat** [comment]
Hi,

I have rebased again
<https://github.com/cloud-bulldozer/benchmark-operator/pull/569/commits/ce48cf00e68cd9ccf125a2076435755883f439c5>
and resolved the conflict.
Can you please rerun it again ?

Thanks,
Eli Battat.

**Comment by ebattat** [comment]
Hi,

Can we proceed with the merge of 2 branches ? (Must be together)
https://github.com/cloud-bulldozer/benchmark-operator/pull/569
https://github.com/cloud-bulldozer/benchmark-wrapper/pull/268

Thanks,
Eli Battat.

**Pull request merged**